mautic/mautic
# Segment delegate decorator event

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [N]
| New feature/enhancement? (use the a.x branch)      | [Y]
| Deprecations?                          | [N]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [Y] 
| Related user documentation PR URL      | /
| Related developer documentation PR URL | /
| Issue(s) addressed                     | /



#### Description:

This PR adds new event that plugins can use to decorator is delegated segment filters. It allows segment filters to be based on objects other than lead and company. This is needed to support the Custom Objects plugin.

#### Steps to test this PR:

Just make sure the segment filters work the same way as before. There should be no change to the UI or behavior.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10890"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
